src/redux/selectors.ts

```diff
@@ -57,11 +59,138 @@ export const isInPreviewModeSelector = createSelector(
 
 export const isInClusterModeSelector = createSelector(
   (state: RootState) => state,
-  appState =>
-    Boolean(appState.main.previewResourceId && appState.main.previewResourceId.endsWith(appState.config.kubeconfigPath))
+  ({main, config}) => {
+    const kubeConfigPath = config.projectConfig?.kubeConfig?.path || config.kubeConfig.path;
+    if (kubeConfigPath) {
+      return Boolean(main.previewResourceId && main.previewResourceId.endsWith(kubeConfigPath));
```

Comment:
Shouldn't this be a boolean by default? since we are using the && operator

Reply:
If the second argument is not boolean AND operator doesn't return boolean

Reply:
true, sorry about that